server/src/main/java/org/elasticsearch/index/fielddata/plain/SortedSetDVOrdinalsIndexFieldData.java

```diff
@@ -127,7 +128,11 @@ public IndexOrdinalsFieldData loadGlobal(DirectoryReader indexReader) {
 
     @Override
     public IndexOrdinalsFieldData localGlobalDirect(DirectoryReader indexReader) throws Exception {
-        return GlobalOrdinalsBuilder.build(indexReader, this, indexSettings, breakerService, logger, scriptFunction);
+        if (indexReader instanceof FilterDirectoryReader){
+           indexReader = FilterDirectoryReader.unwrap(indexReader);
+        }
```

Comment:
Can you add a comment to explain the reasoning? I think we should try to make it more robust by only unwrapping DLS readers. Maybe one way to do that would be to have a sub class of FilterDirectoryReader called something like `UnwrapForGlobalOrdsFilterDirectoryReader` (terrible name, suggestions welcome), make the DLS reader extend it, and do the `instanceof` call on this class here and only unwrap once as opposed to recursively? But that would require that we always wrap with the DLS reader before the FLS reader so that the FLS reader could not hide the DLS reader, I don't know whether that could be problematic.